services/brig/src/Brig/Data/User.hs

```diff
@@ -71,23 +77,37 @@ newAccount u inv = do
     defLoc  <- setDefaultLocale <$> view settings
     uid     <- Id <$> maybe (liftIO nextRandom) (return . toUUID) inv
     passwd  <- maybe (return Nothing) (fmap Just . liftIO . mkSafePassword) pass
-    return (UserAccount (user uid $ locale defLoc) Active, passwd)
+    expiry  <- case status of
+                   Ephemeral -> do
+                       -- Ephemeral users' expiry time is in sessionTokenTimeout seconds
+                       e <- view zauthEnv
+                       let ZAuth.SessionTokenTimeout ttl = e^.ZAuth.settings.ZAuth.sessionTokenTimeout
+                       now <- liftIO =<< view currentTime
+                       return $ Just (addUTCTime (fromIntegral ttl) now)
+                   _ -> return Nothing
+    return (UserAccount (user uid (locale defLoc) expiry) status, passwd)
   where
     ident         = newUserIdentity u
     pass          = newUserPassword u
     name          = newUserName u
     pict          = fromMaybe noPict (newUserPict u)
     assets        = newUserAssets u
+    status        = case ident of
+                        Nothing -> -- any user registering without either an email or a phone is Ephemeral,
+                                   -- i.e. can be deleted after a short amount of time (24h)
```

Comment:
Can we say `sessionTokenTimeout` instead `24h`?

Reply:
sure.